.github/workflows/build-test-pr.yml

```diff
@@ -0,0 +1,51 @@
+name: Gradle Build and Test Pull Requests to OpenHouse
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+
+concurrency:
+  group: ci-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-gradle-project:
+    name: Build Latest commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project sources using merge commit hash
+        uses: actions/checkout@v4
```

Comment:
1. Yes unless the main branch (the merge branch) gets pushed then the merge commit SHA is consistent which we don't want, we want it to checkout the PR Head SHA which is default.
2. Yes the original workflow will only run post-merge, I validated this on my own repository https://github.com/Will-Lo/openhouse/actions. Keeping it the same since there was some behavior so that the image tagging is consistent.